examples/tutorial/jupyter/execution/omnisci_on_native/local/exercise_3.ipynb

```diff
@@ -0,0 +1,146 @@
+   "source": [
+    "When functionality has not yet been implemented for `OmnisciOnNative` execution, we default to pandas as follows\n",
+    "\n",
+    "![](../../../img/omnisci_convert_to_pandas.png)\n",
+    "\n",
+    "We convert the Modin dataframe to a pyarrow.Table, perform a lazy tree execution in OmniSci, render it as a pyarrow.Table, convert it to pandas to perform the operation, and then convert it back to Modin when complete. These operations will have a large overhead due to the communication involved and will take longer than pandas.\n",
+    "\n",
+    "When this is happening, a warning will be given to the user to inform them that this operation will take longer than usual. For example, `DataFrame.apply` is not supported. In this case, when a user tries to use it, they will see this warning:\n",
+	"\n",
+	"```\n",
+	"UserWarning: `DataFrame.apply` defaulting to pandas implementation.\n",
+	"```\n",
+    "\n",
+    "#### Relation engine limitations\n",
+    "As the `OmnisciOnNative` execution is backed by relation algebra based DB engine, there is a certain set of limitations on operations that could be used in Modin with such an execution. For example arbitrary functions in `DataFrame.apply` are not supported as the OmniSci engine can't execute python callables against its tables, this means that `DataFrame.apply(python_callable)` will **always** be defaulting to pandas. \n",
+    "\n",
+    "For more info about `OmnisciOnNative` limitations visit the appropriate section on read-the-docs: [relation algebra limitations](https://modin.readthedocs.io/en/stable/flow/modin/experimental/core/execution/native/implementations/omnisci_on_native/index.html#relational-engine-limitations).\n",
+    "\n",
+    "If your flow mainly operates with non-relational algebra operations, you should better choose non-OmniSci execution (for example, `PandasOnRay`)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Concept for exercise: Default to pandas\n",
+    "\n",
+    "In this section of the exercise we will see first-hand how the runtime is affected by operations that are not implemented."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import modin.pandas as pd\n",
+    "import pandas\n",
+    "import numpy as np\n",
+    "import time\n",
+    "import modin.config as cfg\n",
+    "cfg.StorageFormat.put(\"omnisci\")\n",
+    "\n",
+    "frame_data = np.random.randint(0, 100, size=(2**18, 2**8))\n",
+    "df = pd.DataFrame(frame_data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pandas_df = pandas.DataFrame(frame_data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "modin_start = time.time()\n",
+    "\n",
+    "print(df.apply(sum))\n",
```

Comment:
me and @YarShev checked the perf of modin-default-to-pandas and stock pandas, and it appeared that the default-to-pandas approach is significantly faster! Our suggestion is that the pandas dataframe converted from pyarrow table during the default-to-pandas flow has a different internal chunking structure, which makes this perf difference in comparison with the pandas frame constructed from pydict.

So, the proposal is to replace `.apply` with some other default-to-pandas function that is for sure not faster than stock pandas. Let's use `.mask` as such function, I've checked the perf and it's fallback slower than stock pandas.

Reply:
@dchigarev I changed this